trlx/trainer/accelerate_ppo_trainer.py

```diff
@@ -342,8 +358,10 @@ def make_experience(self, num_rollouts: int = 1024, iter_count: int = 0):  # noq
 
             # store statistics of the initial rollout as reference
             if self.ref_mean is None:
-                self.ref_mean, self.ref_std = scores.mean(), scores.std()
-            all_scores_mean, all_scores_std = self.running_moments.update(scores)
+                self.ref_mean, self.ref_std = (scores * scores_mask).sum(dim=1).mean(), (scores * scores_mask).sum(
```

Comment:
Do you want to take here sum across tokens, or just mean i.e. `.mean(dim=1)`? Although you're not using any type of score normalization in your example, for the case when it's used (`scores /= self.ref_std`) taking mean instead of sum to compute `self.ref_std` should maybe work better here, if I understand correctly

Reply:
Nit: you can also precompute `scores * scores_mask` and pass it later into `self.running_moments.update`

Reply:
I also wasn't sure whether it's better to normalize by returns or rewards. This [paper](https://arxiv.org/pdf/2105.05347.pdf) suggests normalizing by returns is better (at least when using multiple reward scales) so I went with that. Also the reward mean estimate is sensitive to the trajectory length, which I think we want to avoid.

Reply:
> Nit: you can also precompute `scores * scores_mask` and pass it later into `self.running_moments.update`

Good point, I will change this.